src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -21,21 +19,30 @@ RandomFourierGaussPreproc::RandomFourierGaussPreproc()
 	    ParameterProperties::HYPER | ParameterProperties::CONSTRAIN,
 	    SG_CONSTRAINT(positive<>()));
 	SG_ADD(
-	    &m_log_width, "log_width", "Kernel width in log domain",
-	    ParameterProperties::HYPER | ParameterProperties::GRADIENT);
+	    &m_kernel, "kernel", "Kernel object",
+	    ParameterProperties::HYPER);
 	SG_ADD(&m_basis, "basis", "Matrix of basis vectors");
 	SG_ADD(&m_offset, "offset", "offset vector");
 }
 
+RandomFourierGaussPreproc::RandomFourierGaussPreproc(const std::shared_ptr<Kernel>& kernel)
+: RandomFourierGaussPreproc()
+{
+	require(
```

Comment:
move to a `set_kernel` method and call that here

Reply:
I don't see much use of this ctor then

Reply:
Plus, all the other preprocessor only have default constructor. So let's go with the flow?

Reply:
all this is just used for testing anyways. But your call, either remove or move to method

Reply:
Removing this ctor :)